code/datums/mood.dm

```diff
@@ -0,0 +1,468 @@
+#define MINOR_INSANITY_PEN 5
+#define MAJOR_INSANITY_PEN 10
+
+/**
+ * Mood datum
+ *
+ * Contains the logic for controlling a living mob's mood and sanity.
+ */
+/datum/mood
+	/// Weakref to the parent (living) mob
+	var/datum/weakref/parent
+
+	/// Mob's mood
+	var/mood
+	/// Mob's sanity
+	var/sanity = SANITY_NEUTRAL
+	/// The displayed mood
+	var/shown_mood
+	/// Modifier to allow certain mobs to be less affected by moodlets
+	var/mood_modifier = 1
+	/// Used to track what stage of moodies they're on
+	var/mood_level = 5
+	/// To track what stage of sanity they're on
+	var/sanity_level = SANITY_LEVEL_NEUTRAL
+	/// Is the owner being punished for low mood? if so, how much?
+	var/insanity_effect = 0
+	/// The screen object for the current mood level
+	var/atom/movable/screen/mood/mood_screen_object
+
+	/// List of mood events currently active on this datum
+	var/list/mood_events
+
+/datum/mood/New(mob/living/mob_to_make_moody)
+	if (!istype(mob_to_make_moody))
+		stack_trace("Tried to apply mood to a non-living atom!")
+		qdel(src)
+		return
```

Comment:
stack_trace() is meant if you want to CRASH() while also continuing the proc. Since we're just returning, why not CRASH() instead?

Reply:
because i want to qdel the mood datum and THEN return